### DISCUSSION TITLE
Boot Device Not Found

### DISCUSSION THREAD

**Discussion by CDUCC** [ltsp/ltsp#219] [Q&A]
Since I was getting errors with LTSP5 when the clients tried to boot (squashfs errors) after updating Ubuntu, I figured maybe LTSP5 no longer works. So, I tried installing LTSP 19 with the instructions at https://ltsp.org/docs/installation/, but now the clients get this error:
PXE-E78: Could not locate boot server

(Also, as a secondary issue, I'm seeing two lines that read "Boot from network". I've tried both, and get the same error above. But, I'd like to eliminate being asked to choose one of those lines, if possible.)

I'm fairly new to Linux, and don't know how to troubleshoot this.

Thanks,
Carl

**Comment by alkisg** [comment]
Hi Carl,

please upload the following files:

- /srv/tftp/ltsp/ltsp.ipxe
- /etc/dnsmasq.d/ltsp-dnsmasq.conf

And the output of the following commands:

```shell
sudo ltsp info
sudo /usr/lib/klibc/bin/ipconfig -n enp2s0
```

Replace enp2s0 with the name of your NIC; you can see it with `ip a`.

**Comment by enaut** [comment]
alskisg: I did try to reinstall ubuntu with the same error... (might be the same)
pxe did load but then vmlinuz was missing.
`ltsp kernel` did error with missing vmlinuz too.
Is it because of sepparate boot partition?
sorry the short answer going to class now...

**Comment by alkisg** [comment]
Thank you enaut,

I think your issue is caused by a separate boot partition and is a different one than Carl's.

Nevertheless, the stable PPA had LTSP 19.11;
I just copied LTSP 19.12 from the proposed PPA to the stable PPA;
it should be available in half an hour;
let's see if this makes things better.

**Comment by CDUCC** [comment]
> ```shell
> ltsp info
> ```

**Comment by CDUCC** [comment]
ltsp info:
```
LTSP version: 19.11-1~201911211521~ubuntu18.04.1

SERVER XSESSIONS: ubuntu-communitheme-snap.desktop  ubuntu.desktop

SERVER OS: PRETTY_NAME="Ubuntu 18.04.3 LTS"

CHROOTS:


VMs:


IMAGES:
x86_64
```

sudo /usr/lib/klibc/bin/ipconfig -n enp1s0 (Regular Network NIC):
```
IP-Config: enp1s0 hardware address 8c:89:a5:de:85:90 mtu 1500 DHCP RARP
IP-Config: enp1s0 guessed broadcast address 192.168.1.255
IP-Config: enp1s0 complete (dhcp from 192.168.1.1):
 address: 192.168.1.196    broadcast: 192.168.1.255    netmask: 255.255.255.0   
 gateway: 192.168.1.1      dns0     : 192.168.1.1      dns1   : 0.0.0.0         
 domain : localdomain                                                     
 rootserver: 192.168.1.1 rootpath: 
 filename  : 
```

sudo /usr/lib/klibc/bin/ipconfig -n enxd8eb97b0ec4c (LTSP NIC):
`IP-Config: enxd8eb97b0ec4c hardware address d8:eb:97:b0:ec:4c mtu 1500 DHCP RARP`
The LTSP NIC output above never progresses beyond that first line. I've been waiting for a while now, but it seems hung up. So, aborted and tried again, but it's not getting past the first line of output.

**Comment by CDUCC** [comment]
Sorry, forgot the files. Here they are.
[CDUCC.zip](https://github.com/ltsp/community/files/3938208/CDUCC.zip)

**Comment by alkisg** [comment]
> Also, as a secondary issue, I'm seeing two lines that read "Boot from network". I've tried both, and get the same error above.

Can you paste a screenshot of that one? I only see a single entry in your ltsp.ipxe:

> item --gap                        Boot an image from the network in LTSP mode:
> item --key 1 x86_64               x86_64.img

I wonder if you have leftover configuration from ltsp5 that interferes. Do you have any other files in /etc/dnsmasq.d, except for ltsp-dnsmasq.conf?

> The LTSP NIC output above never progresses beyond that first line.

This is correct. The internal LTSP NIC should have a static IP of 192.168.67.1 and shouldn't have any other DHCP servers in that network.
Did you configure your internal NIC to a static IP of 192.168.67.1?

**Comment by CDUCC** [comment]
Yes, you are correct. I did have other files in the /etc/dnsmasq.d folder. I've removed them, and now I don't have to pick anything at startup, but it still fails to boot, as you can see in the image below.

Yes, the LTSP NIC is set statically to 192.168.67.1

![image](https://user-images.githubusercontent.com/58677586/70416282-b96c7f00-1a2c-11ea-80e3-951239015b35.png)

**Comment by alkisg** [comment]
It seems like a TFTP issue.
What is the output of these commands on the server?

```shell
busybox tftp -g -r /ltsp/undionly.kpxe -l - localhost | md5sum
md5sum /srv/tftp/ltsp/undionly.kpxe
```

**Comment by CDUCC** [comment]
> busybox tftp -g -r /ltsp/undionly.kpxe -l - localhost | md5sum

8e76d17840a30383105ae6e34d1f619b  -

> md5sum /srv/tftp/ltsp/undionly.kpxe

8e76d17840a30383105ae6e34d1f619b  /srv/tftp/ltsp/undionly.kpxe

**Comment by alkisg** [comment]
It feels like you still have leftovers from LTSP5.
Could you come online in the [#ltsp IRC channel](http://ts.sch.gr/repo/irc/) to solve this via VNC?
It'll be so much faster.

**Comment by alkisg** [comment]
The problem was UFW (firewall) running on the server, breaking TFTP server,
and then running on the client, breaking NFS client.

So LTSP administrator that enable UFW need to make sure that leave the necessary ports open for at least TFTP and NFS, both server and client.

**Comment by CDUCC** [comment]
Wow, thank you so much for your efforts. I appreciate it!

Carl
